### DISCUSSION TITLE
TabbyML plugin.  I can't get "<Tab>" to auto-complete.

### DISCUSSION THREAD

**Discussion by paulodelgado** [LazyVim/LazyVim#2799] [Q&A] [answered]
TabbyML is a self-hosted OSS alternative to GitHub copilot.  

I've added `tabby.lua` to my `~/.config/nvim/lua/plugins/` directory.  

This is the contents of the file:
```
vim.g.tabby_keybinding_accept = "<Tab>"

return {
  { "TabbyML/vim-tabby", lazy = false },
}
```

The server is running on a separate tab, I can see the code suggestions as I type, but when I hit "<Tab>" all I get is a normal "tab" and autocomplete doesn't happen.    I used to have the snippet above in my `init.lua` before I installed LazyVim and it was working fine. 

Any idea what I'm doing wrong? I feel like LazyVim is overriding the line `vim.g.tabby_keybinding_accept = "<Tab>"`

**Comment by dpetka2001** [comment, marked as answer]
I believe `<Tab>` is used by Luasnip as you can see [here](https://github.com/LazyVim/LazyVim/blob/70bc8803306ba26a6362d489751da48633c1164c/lua/lazyvim/plugins/coding.lua#L103-L114). To override those keymaps you could do something like
```lua
-- ~/.config/nvim/lua/plugins/luasnip.lua
return {
  {
    "L3MON4D3/LuaSnip",
    keys = function()
      -- stylua: ignore
      return {
        { "<C-j>", function() return require("luasnip").jumpable(1) and "<Plug>luasnip-jump-next" or "<tab>" end,
          expr = true,
          silent = true,
          mode = "i",
        },
        { "<C-j>", function() require("luasnip").jump(1) end, mode = "s", },
        { "<C-k>", function() require("luasnip").jump(-1) end, mode = { "i", "s" }, },
      }
    end,
  },
}
```
Then `<Tab>` shouldn't be mapped to anything and I think you could use it as you'd like to. Feel free to change `<C-j>/<C-k>` to anything you feel comfortable with.

**Comment by paulodelgado** [comment]
Wonderful.  That did it! thanks so much.
